### DISCUSSION TITLE
stack creation fail

### DISCUSSION THREAD

**Discussion by ssm2017** [mozilla/hubs#3884] [Hubs Cloud Questions] [answered]
hello
i have deleted a test working stack and the marketplace subscription.
then i have deleted some things in amazon consoles (to not beeing charged for something not used).
our domain is hosted on route 53.
i have deleted all the CNAME entries.
i have deleted the default vpc for my region and created a new default one.

now, i am not able to create a stack anymore.
there is no event listed in ec2.

i have disabled rollback and then it is saying : 
The following resource(s) failed to create: [StackTopicHandlerRole, RouteTablePublicBInternetRoute, RouteTableAssociationAPrivate, SubnetNetworkAclAssociationBPublic, RouteTableAssociationBPrivate, InternalZoneSSLCert, AppDb, SpeelycaptorScratchBucket, AssetsBucket, NearsparkThumbnailLambdaPermissionApiGateway, SESDomainRecords, BoxKeysBucket, SubnetNetworkAclAssociationAPublic, RouteTablePublicAInternetRoute, RootRedirectorBucket, RouteTableAssociationAPublic, SubnetNetworkAclAssociationBPrivate, RouteTableAssociationBPublic, SendEmailUser, SubnetNetworkAclAssociationAPrivate, LinkRedirectorBucket].

do you know if there is a way to reset all my aws parameters and revert them back like i just registered ? :) (to be able to create a new stack)
or, do you know where to look at ?

**Comment by ssm2017** [comment]
after trying to clean vpc (with an haxe) i have new errors (network looks fine this time :) : 
Status reason
The following resource(s) failed to create: [KeymasterRole, SpeelycaptorRolePolicy, EmptyBucketRole, AppDb, SESDomainRecords, StorageEFSAppMountTargetBPrivate, SpeelycaptorApplication, StorageEFSAppMountTargetAPrivate, SendEmailUser].

less errors, that's better

**Comment by robinkwilson** [comment]
Thank you for the list of actions you've taken!

**Can you scroll ALL the way down the events list to where the _first red_ error appears?** It should be specific with what went wrong, then screenshot the error text and paste here.

**Reply by ssm2017** [reply]
hello
thank you for attention
i have tested to create maybe 20 stacks on yesterday and i could run one :) but then deleted it and could not run another one...
i understand now where to look at  (i could read that i have to go to ec2 event list).

the first red error in the cloudformation stack event is saying that i have reached the 20 certs limit...
"InternalZoneSSLCert : Failed to create resource. An error occurred (LimitExceededException) when calling the RequestCertificate operation: Error: you have reached your limit of 20 certificates in the last year."

the hubs doc is saying to request a new limit but i do not know how to request it : 
 * using the quotas manager or using the ticket system (free plan) ?
 * what kind of certificate should i request ?
   * number of acm certs ?
   * number of certificates autorities ?
   * number of privates certificates issued per private ca ?
 * ~~in what region ? (my stack would run in ireland and emails in n.virginia).~~ the region where the stack is created
 * how many certs ? 200 ?
 * ~~is there a workaround to not have to wait few days for their eventual agreement?~~ restriction is per region so i could install a stack on another region during the agreement process

**Reply by ssm2017** [reply]
hello @robinkwilson. i have 3 requests :
 * 1/ is it possible to delete this thread ? (it is mostly noise now)
 * 2/ is it possible to update the docs about the way to find the first "red" issue when rollback is complete ? (the doc is saying "Navigate to AWS EC2 > Select "Events" Tab on left sidebar" but when we create the stack, the ec2 events are empty. so i had to disable rollback and then, after error, i went to the "cloud formation" stack info and go to the "events" tab that has only one line saying "rollback complete" and then go to the "stack info" tab and come back to the "events" tab that is filled now with errors (maybe an amazon bug that is not filling the events on the first view of the tab)...
 * 3/ is it possible to update the docs "Certificate limit reached, must ask AWS for increase" to say precisely what kind of cert to ask for new aws users because there are a lot of certs types requests and it is not clear. on my side, i have requested this :
 "Limit increase request 1. Service: Certificate Manager. Region: EU (Ireland). Limit name: Number of ACM certificates. New limit value: 200"
 but i am not sure if i have asked the good product :)

**Reply by robinkwilson** [reply]
1/ I'd rather keep it available, I'll make the next comment have your answer (actually you've already solved it. After the limit increase, you'll be able to successfully deploy the stack)
2/ THANK YOU!! I'm overjoyed you said something. They are now updated with the correct instructions. I'm so happy you pointed this typo out! https://hubs.mozilla.com/docs/hubs-cloud-aws-troubleshooting.html
3/ That was the correct certificates to increase!!

**Comment by robinkwilson** [comment, marked as answer]
@ssm2017 Already answered their own question.

The error is: "InternalZoneSSLCert : Failed to create resource. An error occurred (LimitExceededException) when calling the RequestCertificate operation: Error: you have reached your limit of 20 certificates in the last year."

They requested an ACM Certificates increase from AWS in their region. 

From @ssm2017 's comment above

> I have requested this :
>"Limit increase request 1. Service: Certificate Manager. Region: EU (Ireland). Limit name: Number of ACM certificates. New limit value: 200"

If another deployment issue pops up, please make a new thread with the new error. Thanks again for your help with the docs!!

**Reply by robinkwilson** [reply]
(Hopefully this is succinct for an answer @ssm2017)
Also if you'd like to be selected as the answerer, feel free to paraphrase in a new "Suggest and answer" and I'll select your answer.
